### PR TITLE
Brute force version with tests

### PR DESCRIPTION
**Pull request by ghitafilali**
Really sorry again for the late submission. I finally got a functional version of the brute force method and I implemented all the tests that seemed relevant.

### CODE REVIEW

File: ghitafilali/assignment-3/word_search.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        return word in self.words
+
+    def is_prefix(self, prefix):
+        """
+        Returns whether or not there exists a word in the lexicon with
+        the prefix prefix.
+        """
+        for word in self.words:
+            prefix_len = len(prefix)
+            if len(word) < prefix_len:
```
Comment (kevmo314) [inline ghitafilali/assignment-3/word_search.py]: This conditional shouldn't be necessary, as the string equality operator checks length before attempting to compare too.

File: ghitafilali/assignment-3/word_search.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        if not remaining:
+            return True
+        current_letter = remaining[0]
+        for neighbour in neighbours:
+            if neighbour not in visited_letters:
+                if self.grid[neighbour[0]][neighbour[1]] == current_letter:
+                    letter_neigh = self.get_neighbour_coordinates(neighbour[0],
+                                                                  neighbour[1])
+                    new_visited_letters = visited_letters.copy()
+                    new_visited_letters.add(neighbour)
```
Comment (kevmo314) [inline ghitafilali/assignment-3/word_search.py]: You can create a new set by doing `visited_letters | set([neighbor])`: https://docs.python.org/3/library/stdtypes.html#frozenset.union

File: ghitafilali/assignment-3/word_search.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        """
+        Recursive function that returns whether the remaining letters can be
+        found by moving to one of the 8 adjacent cells without visinting twice
+        the same cell.
+        """
+        if not remaining:
+            return True
+        current_letter = remaining[0]
+        for neighbour in neighbours:
+            if neighbour not in visited_letters:
```
Comment (kevmo314) [inline ghitafilali/assignment-3/word_search.py]: To reduce the indentation, switch the direction of this conditional:

```
if neighbor in visited_letters:
  continue
```

and same for the next conditional.

File: ghitafilali/assignment-3/word_search.py (hunk 1 above)
Comment (ghitafilali) [reply ghitafilali/assignment-3/word_search.py]: Okay I'll remove it then

File: ghitafilali/assignment-3/word_search.py (hunk 2 above)
Comment (ghitafilali) [reply ghitafilali/assignment-3/word_search.py]: Okay thanks

File: ghitafilali/assignment-3/word_search.py (hunk 3 above)
Comment (ghitafilali) [reply ghitafilali/assignment-3/word_search.py]: Okay I'll change it, I've never actually used "continue" but now I know how it works, so thanks !

**Pull request merged**